InventoryConfigurableProductIndexer/Indexer/SelectBuilder.php

```diff
@@ -100,6 +115,10 @@ public function execute(int $stockId): Select
                 ['parent_product_entity' => $this->resourceConnection->getTableName('catalog_product_entity')],
                 'parent_product_entity.' . $linkField . ' = parent_link.parent_id',
                 []
+            )->joinInner(
+                ['parent_stock' => $this->resourceConnection->getTableName('cataloginventory_stock_item')],
```

Comment:
Is it possible to avoid `cataloginventory_stock_item ` table usage? CatalogInventory module will be deprecated and removed in the next releases, we should use Inventory logic only to handle such cases.